packages/lodestar/src/network/reqresp/reqResp.ts

```diff
@@ -154,7 +159,9 @@ export class ReqResp implements IReqResp {
 
       return result;
     } catch (e) {
-      const peerAction = onOutgoingReqRespError(e as Error, method);
+      this.metrics?.reqRespOutgoingErrors.inc({method});
```

Comment:
I would track dial errors here, puting the logic inside `onOutgoingReqRespError` breaks the principle of "funcitons should do one thing".

```ts
if (e instanceof RequestError && e.type.code === RequestErrorCode.DIAL_ERROR) {
  metrics?.reqRespDialErrors.inc();
}

```

Reply:
DAIL_TIMEOUT not to be counted in reqRespDialErrors?

Reply:
Oh! Right that too